feat: sentry integration

soren5 opened the pull request:
Followed instructions from:
https://github.com/getsentry/sentry-laravel
https://docs.sentry.io/clients/php/integrations/laravel/

to configure sentry and added some instructions about setup in the readme.md

Also implemented sentry reporting for uncaught api exceptions

soren5 committed:
chore: configure sentry

create necessary files and configurations to integrate sentry

Issue #10

soren5 committed:
feat: add sentry reporting

report uncaught api exceptions using sentry

Issue #10

hmatalonga commented on app/Exceptions/Handler.php:
```diff
@@ -39,6 +39,10 @@ class Handler extends ExceptionHandler
      */
     public function report(Exception $exception)
     {
+        if (app()->bound('sentry') ) {
```
I was wondering why did you not include `&& $this->shouldReport($exception)` to the if statement?

soren5 replied:
When I made some testing I noticed that $this->shouldReport did not return anything for error 401 so I thought we needed to define shouldReport through some middleware.

After further testing I believe that when an exception should NOT be reported shouldReport will return nothing (which evaluates as false) and shouldntReport will return 1 (which evaluates as true)

Screenshot of logs for exception 401:

<img width="682" alt="screen shot 2018-09-10 at 11 32 33" src="https://user-images.githubusercontent.com/25614957/45292507-591b2480-b4ed-11e8-9243-714411dc368e.png">

<img width="553" alt="screen shot 2018-09-10 at 11 30 32" src="https://user-images.githubusercontent.com/25614957/45292492-50c2e980-b4ed-11e8-90da-9271fea8c4e7.png">

In conclusion I'll be using shouldReport in the next push

hmatalonga commented on app/Exceptions/Handler.php:
```diff
@@ -83,16 +85,13 @@ public function render($request, Exception $exception)
-                ], 500);
+                break;
         }
+        $exception = new HttpException(500, 'Whoops!');
```
What was the reason to create a new exception here?

Maybe we could just do?
```php
return parent::render($request, $exception);
```

soren5 replied:
Yes I believe this is correct, I'll include it in the next push

hmatalonga reviewed (changes requested):
Looks good to me :+1: 

Just a couple of minor details and this will be merged!

soren5 committed:
refactor: improve reporting conditions

restrict condition for sentry reporting
remove unnecessary exception creation
fix some incorrect function calls

Issue #10

hmatalonga commented on app/Exceptions/Handler.php:
```diff
@@ -57,7 +57,7 @@ public function report(Exception $exception)
     {
         if (!$request->is("api/*")) {
-            return parent::render($request, $exception);
+            return parent::report($request, $exception);
```
Was there a reason to invoke report? I didn't understand the change...

This line looked fine, it is the `render` function so the call to the parent method was correct.

soren5 replied:
I didn't realize that each exception called both render and report so I made the change to make sure all necessary exceptions got reported.
I looked at some other examples of exception handling and I think both methods will get called every time so the change was unnecessary

hmatalonga commented on app/Exceptions/Handler.php:
```diff
@@ -88,8 +88,7 @@ public function render($request, Exception $exception)
         }
-        $exception = new HttpException(500, 'Whoops!');
-        parent::render($request, $exception);
+        parent::report($exception);
```
I was referring to delegate the default case to the parent just like so
```php
return parent::render($request, $exception);
```

soren5 committed:
fix: undo incorrect method call

change return value of render function

hmatalonga commented:
Closes #10
